examples/developer-tools/src/output-generators/json_definition_generator.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+/**
+ * We will never need parentheses for the JSON definition, so
+ * order precedence can be ignored.
+ */
+export const Order = {
+  ATOMIC: 0,
+};
+
+/**
+ * A CodeGenerator that generates a JSON block definition for a block defined
+ * in the Block Factory.
+ */
+export class JsonDefinitionGenerator extends Blockly.CodeGenerator {
+  /**
+   * Creates a new JsonDefinitionGenerator.
+   *
+   * @param name Registered name for the generator.
+   */
+  constructor(name = 'JsonBlockDefinition') {
+    super(name);
+    // this.isInitialized = false;
+
+    // TODO: List any reserved words.
+    // this.addReservedWords();
+  }
+
+  /**
+   * Initialise the database of variable names.
+   *
+   * @param workspace Workspace to generate code from.
+   */
+  init(workspace: Blockly.Workspace) {
```

Comment:
Do we want commented out code here?  Over time the origin of this code will drift away from the commented out code.  I'm not sure it can be trusted in the future.

Reply:
No, we don't. I'll remove it all before I merge this branch into develop. I just haven't gotten far enough along to where I know which parts I want to delete and which parts I actually want to uncomment. But I won't do the final merge with this in here